api/routes/subcriterias/get_sub_criteria.py

```diff
@@ -43,3 +49,160 @@ def return_subcriteria_from_config(sub_criteria_id):
+    def get_themes_answers(cls, theme_id: str) -> list[dict]:
+        """function takes a theme_id arg & returns a list
+        of answers with given theme_id.
+        """
+        sub_criterias = get_all_subcriteria()
+        try:
+            return [
+                theme.get("answers")
+                for themes in sub_criterias
+                for theme in themes.get("themes")
+                if theme_id == theme.get("id")
+            ][0]
+        except IndexError:
+            current_app.logger.error(f"Incorrect theme id -> {theme_id}")
+            return f"Incorrect theme id -> {theme_id}"
+
+    @classmethod
+    def get_application_form(cls, app_json_blob):
+        """function return list of all questions from application form"""
+
+        return [
+            questions
+            for forms in app_json_blob["jsonb_blob"]["forms"]
+            for questions in forms["questions"]
+        ]
+
+    @classmethod
+    def convert_boolean_values(cls, themes_answers: list[dict]) -> list[dict]:
+        """function checks boolean values in themes_answers & replace
+        boolean values to string. (False -> "No", True -> "Yes")
+        Args:
+            themes_answers (_type_): array of dict
+        """
+        current_app.logger.info("Converting boolean values to strings")
+        for answers in themes_answers:
+            if "answer" in answers.keys():
+                if answers["answer"] == False:
+                    answers.update(answer="No")
+                if answers["answer"] == True:
+                    answers.update(answer="Yes")
+
+    @classmethod
+    def sort_add_another_component_contents(
+        cls, themes_answers: list[dict]
+    ) -> list[dict]:
+        """function checks for special presentation_type "heading"
+        in array of themes answers, if exists, adds question-value
+        to an answer key for presentation_type "heading" theme and
+        looks for presentation_type of "description" and "amount" with
+        same presentation_type "heading"s field_id, retrieve words only
+        from strings of answers for presentation_type "description" theme
+        & numbers only for presentation_type "amount" theme.
+
+        Args:
+            themes_answers (_type_): array of dict
+        """
+        for heading in themes_answers:
+            try:
+                if heading["presentation_type"] == "heading":
+                    current_app.logger.info(
+                        "mapping add-another component contents"
+                    )
+                    heading["answer"] = heading["question"]
+                    for theme in themes_answers:
+                        if (
+                            heading["field_id"] in theme.values()
+                            and theme["presentation_type"] == "description"
+                        ):
+                            description_answer = [
+                                re.sub("[^a-zA-Z]+", " ", description)
```

Comment:
What are we manipulating in this regex?

Reply:
Got rid of regex & now using split method to get the description